news-explorer-frontend/src/components/App/App.jsx

```diff
@@ -0,0 +1,342 @@
+import { CurrentUserContext } from '../../contexts/CurrentUserContext';
+import react, { useState, useEffect} from 'react';
+import { useLocation, useNavigate, Routes, Route, Router, Navigate } from 'react-router-dom';
+import './App.css';
+import '../../index';
+import Main from '../Main/Main';
+import Header from '../Header/Header';
+import Footer from '../Footer/Footer';
+import SignInPopup from '../Signin/Signin';
+import SignUpPopup from '../SignUp/Signup';
+import RegisterationSuccess from '../Success/Success';
+import MobileSignIn from '../MobileSignIn/MobileSignIn';
+import MobileMenu from '../MobileMenu/MobileMenu';
+import api from '../../utils/MainApi';
+import newsApi from '../../utils/NewsApi';
+import { data } from '../../utils/data';
+import ProtectedRoute from '../ProtectedRoute/ProtectedRoute';
+import SavedArticles from '../SavedArticles/SavedArticles';
+
+
+function App() {
+  const navigate = useNavigate();
+  const [currentUser, setCurrentUser] = useState({});
+  const [savedArticle, setSavedArticle] = useState ([]);
+  const [cards, setCards] = useState([]);
+  const [userData, setUserData] = useState({ name: 'name'});
+  const [isCheckingToken, setIsCheckingToken] = useState(true)
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isSignInPopupOpen, setIsSignInPopupOpen] = useState(false);
+  const [isSignUpPopupOpen, setIsSignUpPopupOpen] = useState(false);  // isLoggedIn & isSignUpPopupOpen
+  const [isRegisterPopupOpen, setIsRegisterPopupOpen] = useState(false);
+  const [isMobileClicked, setIsMobileClicked] = useState(false);
+  const [isMobileMenuClicked, setIsMobileMenuClicked] = useState(false); 
+  const [filteredResults, setFilteredResults] = useState([]);
+  const [keyword, setKeyword] = useState('');
+
+    //MainApi signup
+    const onRegisterUser = ({ name, email, password }) => {
+      api.signup({name, email, password})
+        // .then((res) => {
+        //   if(res._id) {
+        //     navigate('/signin')
+        //   }
+        // })
+        // .then(() => {
+        //   setIsSignInPopupOpen(false)
+        //   setIsRegisterPopupOpen(true)})
+        .catch((err) => {
+          console.log("signup err =>", `Error: ${err.status}`);
+        })
+        .finally(() => {
+          closeAllPopups();
+          setIsRegisterPopupOpen(true);
+        })
+    }
+
+    // MainApi Signin
+    const onLogin = ({  email, password }) => {
+      api.signin({email, password})
+        .then((res) => {
+          if(res.token) {
+            setIsLoggedIn(true);
+            setUserData(res.name); // check if I need to use this later on
+            localStorage.setItem('jwt', res.token);
+            navigate('/');
+          } else {
+            console.log("signin fail")
+          }
+        })
+        .catch((err) => {
+          console.log("signin fail=>", err);
+        })
+        .finally(() => closeAllPopups())

```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well


The user should see what he/she failed logging in or you shouldn't close it in `finally`

Reply:
fixed++